src/consensus/proof.rs

```diff
@@ -75,20 +77,28 @@ impl Debug for ProofShare {
 
 /// A value together with the proof that it was agreed on by the quorum of the section elders.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Serialize, Deserialize)]
-pub struct Proven<T> {
+pub struct Proven<T: Serialize> {
     pub value: T,
     pub proof: Proof,
 }
 
-impl<T> Proven<T> {
+impl<T: Serialize> Proven<T> {
     pub fn new(value: T, proof: Proof) -> Self {
         Self { value, proof }
     }
+
+    pub fn self_verify(&self, history: &SectionProofChain) -> bool {
```

Comment:
Very minor point: maybe this should no longer be called `self_verify` as it does the trusted keys check too? Maybe call it just `verify`?.